### PR TITLE
Abdul/tk slot indexing

### PR DESCRIPTION
**Pull request by imabdulbasit**

### CODE REVIEW

File: crates/core/migrations/2022-03-22-214148_add_slot_col_and_trigger_to_token_account/up.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+begin
+    new.updated_at = now();
+    return new;
+end;
+$$ language 'plpgsql';
+
+create trigger set_token_account_updated_at
+before update on token_accounts
+for each row
+execute procedure trigger_set_updated_at_timestamp();
```
Comment (kespinola) [inline crates/core/migrations/2022-03-22-214148_add_slot_col_and_trigger_to_token_account/up.sql:17]: nice.

**Review by kespinola — COMMENTED** [review]
Looking good. Have some nested if statements maybe @ray-kast has some technics for handling.

File: crates/indexer/src/accountsdb/accounts/token.rs [hunk 2]
```diff
@@ -22,26 +25,66 @@ pub async fn process(client: &Client, key: Pubkey, token_account: TokenAccount)
...
+            .db()
+            .run(move |db| {
+                insert_into(token_accounts::table)
+                    .values(&values)
+                    .on_conflict_do_nothing()
+                    .execute(db)
+            })
+            .await
+            .context("failed to insert token account")?;
+    }
```
Comment (ray-kast) [inline crates/indexer/src/accountsdb/accounts/token.rs:87]:
```suggestion
    match rows.get(0).and_then(|r| r.slot) {
        Some(indexed_slot) if incoming_slot > indexed_slot => {
            client
                .db()
                .run(move |db| {
                    update(
                        token_accounts::table
                            .filter(token_accounts::address.eq(values.clone().address)),
                    )
                    .set(&values)
                    .execute(db)
                })
                .await
                .context("failed to update token account")?;
        },
        Some(_) => (),
        None => {
            client
                .db()
                .run(move |db| {
                    insert_into(token_accounts::table)
                        .values(&values)
                        .on_conflict_do_nothing()
                        .execute(db)
                })
                .await
                .context("failed to insert token account")?;
        },
    }
```

**Review by ray-kast — CHANGES_REQUESTED** [review]

File: crates/indexer/src/accountsdb/accounts/token.rs [hunk 3]
```diff
@@ -22,26 +25,62 @@ pub async fn process(client: &Client, key: Pubkey, token_account: TokenAccount)
...
+                .context("failed to update token account")?;
+        },
+        Some(_) => (),
+        None => {
+            client
+                .db()
+                .run(move |db| {
+                    insert_into(token_accounts::table)
+                        .values(&values)
+                        .on_conflict_do_nothing()
```
Comment (ray-kast) [inline crates/indexer/src/accountsdb/accounts/token.rs:77]: may want this to be an `ON CONFLICT UPDATE`.  @imabdulbasit thoughts?

**Pull request merged**